src/broker/index.js

```diff
@@ -71,17 +81,29 @@ module.exports = class Broker {
       description: `connect to broker ${this.brokerAddress}`,
     })
 
+    this.authLock = new Lock({
+      timeout: this.authenticationTimeout,
```

Comment:
Let's just add a little 50ms buffer to the lock timeout. The reason being that if there is an authentication timeout, we want the error to come from the authentication timeout and not a generic lock timeout. The lock timeout is just there to ensure that we don't hold a lock forever.

Reply:
good thinking, I'll add it

Reply:
done